File: backend/src/main/java/com/example/demo/entities/Purchase.java
package com.example.demo.entities;


import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import java.util.Date;
import java.util.List;


@Getter
@Setter
@EqualsAndHashCode
@ToString
@Entity
@Table(name = "ordine",schema = "public")
@SecondaryTable(name = "pagamento", pkJoinColumns = {@PrimaryKeyJoinColumn(name = "id")})
public class Purchase {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private int id;

    @Basic
    @CreationTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    @Column(name = "purchase_time")
    private Date purchaseTime;

    @ManyToOne
    @JoinColumn(name = "cliente")
    private User buyer;

    @OneToMany(mappedBy = "purchase", cascade = CascadeType.MERGE)
    private List<ProductInPurchase> productsInPurchase;

    @Basic
    @Column(name = "numero_carta",table = "pagamento",length = 16,nullable = false)
    private String numero_carta;

    @Basic
    @Column(name = "tipo",table = "pagamento",length = 50,nullable = false)
    private String tipo;

    @Basic
    @Column(name = "scadenza",table = "pagamento",nullable = false)
    private String scadenza;


}

File: frontend/lib/Entities/Purchase.dart
import 'package:nuovo_flutter/Entities/Cliente.dart';
import 'package:nuovo_flutter/Entities/ProductInPurchase.dart';
import 'package:nuovo_flutter/Entities/prodotto.dart';

class Purchase{
  List<ProductInPurchase> productsInPurchase;
  cliente buyer;
  int id;
  DateTime purchaseTime;
  String numero_carta;
  String tipo;
  String scadenza;

  Purchase({this.id,this.buyer,this.numero_carta,this.productsInPurchase,this.purchaseTime,this.scadenza,this.tipo});

  Map<String,dynamic> toJson() =>{
    'buyer':buyer.toJson(),
    'id':id,
    'purchaseTime':purchaseTime,
    'numero_carta':numero_carta,
    'tipo':tipo,
    'scadenza':scadenza,
    'productsInPurchase':_printProducts(productsInPurchase),
  };

  List<Map<String,dynamic>> _printProducts(List<ProductInPurchase> cr){
    var ris=<Map<String,dynamic>>[];
    for(ProductInPurchase i in cr){
      ris.add(i.toJson());
    }
    return ris;
  }

}